earth_enterprise/src/fusion/portableglobe/gerewritedbroot.cpp

```diff
@@ -219,21 +225,28 @@ void ReplaceReferencedKml(const std::string &new_kml_base_url,
       if (string_id_or_value->has_value()) {
         // the url is stored directly in the kml_url field
         std::string kml_url = string_id_or_value->value();
-        fout << kml_url << " " << kml_file_name << std::endl;
+        if ( preserve_kml_filenames ) {
+            kml_file_name = kml_file_name.substr(kml_file_name.find_last_of("/") + 1);
+        }
         string_id_or_value->set_value(new_kml_base_url + "/" + kml_file_name);
       } else {
         // the url is in the translation_entry table
         // GetTraslationEntryById will throw if not found
         keyhole::dbroot::StringEntryProto *string_entry =
             dbroot->GetTranslationEntryById(string_id_or_value->string_id());
-        std::string kml_url = string_entry->string_value();
-        fout << kml_url << " " << kml_file_name << std::endl;
+        kml_url = string_entry->string_value();
+        if ( preserve_kml_filenames ) {
+            kml_file_name = kml_file_name.substr(kml_file_name.find_last_of("/") + 1);
+        }
         string_entry->set_string_value(new_kml_base_url + "/" + kml_file_name);
       }
     }
+    // Write out map entry to file if one is specified.   
+    if ( fout ) 
+      fout << kml_url << " " << kml_file_name << std::endl;
   }
-
-  fout.close();
+  if ( fout ) 
```

Comment:
Minor: this is just a matter of preference, but I prefer for if statements to be on a single line (`if ( fout ) fout.close();`) or to have braces.